### PR TITLE
Replies thread slack like

### LINKED ISSUE #1112: Threads
Is there a thread package available that works like Flowdock has that allows people to reply to certain messages and group them?
## <bountysource-plugin>

Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/27434771-threads?utm_campaign=plugin&utm_content=tracker%2F16890630&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F16890630&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by khaledalyawad**
Closes #1112

### CODE REVIEW

**Comment by khaledalyawad** [comment]
I migrated slack like replies to the latest 1.0.0-beta.2

**Comment by mrsimpson** [comment]
Great you give it another shot!
However, you used the code from the old code base, including th `assistify-threading` package. The content of this packages has been moved to `app/threading`. Kindly use and modify the files there. 

Also, it was help to others of you explained here in some word what you change to achieve what.

**Comment by khaledalyawad** [comment]
it is based of the last release https://github.com/RocketChat/Rocket.Chat/commit/63bb92a80a6d0e76983c9d4d63d79f896eea226a

File: .meteor/packages [hunk 1]
```diff
@@ -197,6 +197,7 @@ rocketchat:search
 chatpal:search
 rocketchat:lazy-load
 tap:i18n
+assistify:threading
```
Comment (mrsimpson) [inline .meteor/packages]: This package should not be there anymore - this is now in `/app/discussion`.

File: client/methods/addMessageReply.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
+import { Meteor } from 'meteor/meteor';
```
Comment (mrsimpson) [inline client/methods/addMessageReply.js]: All these files are obsolete and make the changelist longer than it should be

**Comment by mrsimpson** [comment]
Hi @khaledalyawad ,
I had a look at the changelist and found there are many files from an older implementation in the `packages` folder which [don't exist anymore](https://github.com/RocketChat/Rocket.Chat/tree/63bb92a80a6d0e76983c9d4d63d79f896eea226a/packages).

Thus, I tried to help and merge the current `develop`, but this caused a lot of other changes. I wanted to see how it works and where the changes are, so I fixed most of them in 95fa16bbae8aebdc0906745f77825b96bc4c7d54 .

Now, the server technically starts, but I can't get the reply-feature tested. The context menu is now gone.
It may be the result of a wrong merge, so my fault - but you may anyway merge 95fa16bbae8aebdc0906745f77825b96bc4c7d54 into your branch and continue from there.

@ others: You can see the actual footprint [by comparing my commit with develop](https://github.com/RocketChat/Rocket.Chat/compare/develop...95fa16bbae8aebdc0906745f77825b96bc4c7d54)

**Comment by engelgabriel** [comment]
Thank you @khaledalyawad for the hard work on this PR and @mrsimpson for the feedback and deep technical considerations about the underlying data model. We have used this code as the bases for our implementation and the feedback to make improvements to address most of the listed issues and concerns.

The result is the ongoing PR https://github.com/RocketChat/Rocket.Chat/pull/13996 bu @ggazzo which will be marked as coauthored with you both.

Thank you again for helping us build such a complex and highly demanded feature.

**Comment by khaledalyawad** [comment]
@engelgabriel  Thanks, I noticed that my name is not there in the release contributors https://github.com/RocketChat/Rocket.Chat/releases

**Pull request closed without merging**